plugins/outputs/file/file.go

```diff
@@ -39,13 +43,26 @@ func (f *File) SetSerializer(serializer serializers.Serializer) {
 	f.serializer = serializer
 }
 
-func (f *File) Connect() error {
-	writers := []io.Writer{}
-
+func (f *File) Init() error {
+	var err error
 	if len(f.Files) == 0 {
 		f.Files = []string{"stdout"}
 	}
 
+	var options []internal.EncodingOption
+	if f.CompressionLevel >= 0 && f.CompressionAlgorithm != "" && f.CompressionAlgorithm != "identity" {
+		options = append(options, internal.WithCompressionLevel(f.CompressionLevel))
+		f.encoder, err = internal.NewContentEncoder(f.CompressionAlgorithm, options...)
+	} else {
+		f.encoder, err = internal.NewContentEncoder(f.CompressionAlgorithm)
+	}
```

Comment:
I don't get what you are doing here... Why do you need a special treatment for `identity`? You can simply do
```suggestion
	if f.CompressionAlgorithm == "" {
		f.CompressionAlgorithm = "identity"
	}
	var options []internal.EncodingOption
	if f.CompressionLevel >= 0  {
		options = append(options, internal.WithCompressionLevel(f.CompressionLevel))
	}
	f.encoder, err = internal.NewContentEncoder(f.CompressionAlgorithm, options...)
```

Reply:
Because it won't work. You made it error out when given options.
https://github.com/influxdata/telegraf/blame/master/internal/content_coding.go#L326

Reply:
Right, but you not giving it an option if you leave the defaults...

Reply:
Your suggestion is always giving it an option. If you tried it, tests will fail.

Reply:
Is it better? or I'm really clueless on what should be done.